src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx

```diff
@@ -55,15 +82,53 @@ export const WalletConnectDetails = ({ isOpen, wallet, accounts, onClose }: Prop
     return accountsList;
   }, []);
 
+  const reconnect = () => {
+    connect({
+      chains: getWalletConnectChains(chainsService.getChainsData()),
+      pairing: { topic: accounts[0].signingExtras?.pairingTopic },
+    });
+
+    reconnectStarted();
+  };
+
+  const isAccountsStep = reconnectStep === ReconnectStep.NOT_STARTED && connected;
```

Comment:
⚠️ Logic in UI ⚠️
These flags should stay in model which knows how to produce them